chromedp.go

```diff
@@ -385,6 +388,60 @@ func (c *Context) attachTarget(ctx context.Context, targetID target.ID) error {
 	return nil
 }
 
+// Authentication proxy
+func Authentication(login string, password string) Tasks {
+	return Tasks{
+		fetch.Enable().WithPatterns([]*fetch.RequestPattern{{"*", "", ""}}).WithHandleAuthRequests(true),
+		ActionFunc(func(ctx context.Context) error {
+			if login != "" && password != "" {
+				c := FromContext(ctx)
+				if c == nil || c.Allocator == nil || c.cancel == nil {
+					return ErrInvalidContext
+				}
+				id := 1000
+				ListenTarget(ctx, func(ev interface{}) {
+					switch ev := ev.(type) {
+
+					case *fetch.EventAuthRequired:
+						buf, _ := json.Marshal(map[string]interface{}{
+							"requestId": ev.RequestID.String(),
+							"authChallengeResponse": map[string]string{
+								"response": "ProvideCredentials",
+								"username": login,
+								"password": password,
+							},
+						})
+						cmd := &cdproto.Message{
+							ID:        int64(id + 1),
+							SessionID: c.Target.SessionID,
+							Method:    cdproto.MethodType("Fetch.continueWithAuth"),
+							Params:    buf,
+						}
+						err := c.Browser.conn.Write(ctx, cmd)
```

Comment:
It should not write raw messages here. `fetch.ContinueWithAuth` should be used instead.